### ISSUE TITLE
Apply commands to currently selected word

### ISSUE THREAD

**Issue by balta2ar** [t9md/vim-smalls#5]
When I search for something, select some candidate with Tab, and then go to excursion mode and press `y`, `c` or `d`, the action is performed on the whole range from current cursor location till the candidate.

Let me give an example. I have the following text: `|one two three three` (where `|` is cursor). I type (`s` is a binding for vim-smalls): `sthree<Tab>d` -- after that the whole line is deleted. And I would like it to delete only the second occurrence of "three".

**Comment by t9md** [comment]
I understood what you explained, but smalls' functionality is already too rich to original purpose for me.
My original intention for this plugin is simply move position of cursor with seamless and intuitive manner.

Excursion mode is big experiment done for my programming joy, and I don't think its truly necessary.
 
If you want to delete second `three`, move cursor  to 2nd `three` and do `diw` which is vim's native way.

Of course the feature you suggested is good for you and some other people and adding this feature is not difficult ,but  I don't have enough motivation to enhance excursion mode further.

**Comment by balta2ar** [comment]
> If you want to delete second three, move cursor to 2nd three and do diw which is vim's native way.

Sure, that's what I do now. But 1) excursion mode with its already supported excursion commands and highlighting reminds visual mode, and it supports similar commands, however their behavior is surprising to me 2) this feature will let me type less.

> Of course the feature you suggested is good for you and some other people and adding this feature is not difficult ,but I don't have enough motivation to enhance excursion mode further.

OK, I understand you. Do you mind to keep this issue open in case someone (maybe me) implements it in the future?

**Comment by t9md** [comment]
I didn't intend to surprise you :).
How about tying vs after omap(operator mode map), that visuarize affected area. That's my original intention.
But I'm OK to implement your suggestion.
Implementation shoud be for all yank delete change for current word.
Please wait for a short time.

**Comment by balta2ar** [comment]
> How about tying vs after omap(operator mode map), that visuarize affected area. That's my original intention.

I'm sorry, I didn't quite catch it. Could you please elaborate or give an example?

> Please wait for a short time.

Sure, no problem, no hurry. Thank you!

Actually, I have more ideas :) I guess they go beyond your original intention to keep things simple but please hear me out and share your thoughts.

Interactivity in vim-smalls is a killer feature for me. Instant feedback is what I really love about this plugin. I have a feeling that it might be very handy elsewhere. There is absolutely lovely plugin that adds multiple cursors to vim (https://github.com/terryma/vim-multiple-cursors). I use it all the time. I think vim-smalls and vim-multiple-cursors can play nice together, here is how: add command `do_cursors` to turn all highlighted words into multiple cursors (this might need changes to vim-multiple-cursors).

Motivation for this is as follows: vim-multiple-cursors is not interactive/instant/visual enough for me. I often want to preview all the words that I might want to turn to cursors. To do this I either need to select them (pressing `<C-n>` several times) and cancel cursors mode if I didn't like it, use `MultipleCursorsFind` command that takes regular expression as an argument (but it's not instant!), or search them using `/`, which is again not instant and doesn't highlight search results as I type (another option would be to use Mark/vim-quickhl).

Considering above-mentioned, here are the things that are missing:
1. `do_cursors` command that turns highlighted words into cursors.
2. regular expression in the search field. Does vim-smalls support that? I tried `.` and it works but `.*` and `.+` don't.
3. search within visual selection if in Visual mode. I need this to limit the range where I want my cursors to be created (say, only current function).
4. `@/` register is not updated after the search. EasyMotion does update it. What do you think about an option to toggle that?

I know this might sound like too much of a burden/too complicated for vim-smalls. Here are the ways that I see this could be implemented otherwise:
1. Implement `do_cursors` in EasyMotion. Pros are that 1) it instantly highlighs 2) it searches whole file, not just visible area.
2. Implement `TurnSearchQueryIntoCursors` function in vim-multiple-cursors for Normal and Visual modes.

I'd love to hear what you think about all this.

**Comment by balta2ar** [comment]
It turns out it's easy to create cursors from search register: `nnoremap <silent> <M-n> :MultipleCursorsFind <C-R>/<CR>`.

**Comment by t9md** [comment]
I experimentally implemented new action that affect `<cword>`.
You can try by setting following configuration in your vimrc.
This is not action what you exactly wanted, I know.
But, within smalls code, currently highlighted word is not exactly tracked for action purpose although its tracked for interactive feedback.
So adding action which affect exact same area of current highlighted word  need a lot of code change and I still have no motivation for this development.

```Vim
let s:excursion_table_custom = {
      \ "y": "do_yank_cword",
      \ "c": "do_change_cword",
      \ "d": "do_delete_cword",
      \ }
call smalls#keyboard#excursion#extend_table(s:excursion_table_custom)
```
I still don't carefully read your suggestion for multiple cursor combination, 
I came up with my mind you would be impressed some of other plugin like following.
These are also interactive user feedback for your search/replace/change.

https://github.com/haya14busa/incsearch.vim
https://github.com/osyo-manga/vim-over

**Comment by balta2ar** [comment]
> I experimentally implemented new action that affect <cword>.

Thanks! I've tried it and it is indeed better now. However, there is one weird point: it is off by one character. If I have this text `|one two three four`, I search for "three", press `Tab`, then `c`, my cursor is placed here: `one two|  four` (there should be TWO spaces after the cursor, but looks like github formatter eats it), while if I do it vim's native way, cursor is placed here: `one two | four` (one character to the right).

> https://github.com/haya14busa/incsearch.vim

Wow, that's what I've been missing so much! Thank you, I didn't know about this one!

**Comment by t9md** [comment]
The behaviour you described for `c` action is one I already noticed.
Original `c` action in smalls is implemented with combination of `d`(delete) and `:startinsert` to command because of the odd cursor position you reported.
But in this time enhancement, I chose short path to quickly implement `ciw` action via `:normal! ciw` commands.
I don't want to mess up original code and have much motivation, sorry for inconvenience for `change_cword` action, I won't fix it soon.
